common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -329,6 +343,27 @@ private static void appendReplacementCharacter(ArrayList<Byte> bytes) {
    * @return A new UTF8String that is a valid UTF8 string.
    */
   public UTF8String makeValid() {
+    return UTF8String.fromBytes(makeValidBytes());
```

Comment:
```suggestion
    if (isValid) return this;
    return UTF8String.fromBytes(makeValidBytes());
```
I believe we can avoid recomputing validity if we already know the current UTF8 string is valid.